gobblin-core/src/main/java/gobblin/writer/AvroHdfsDataWriter.java

```diff
@@ -89,7 +89,7 @@ public long recordsWritten() {
   }
 
   @Override
-  public long bytesWritten() throws IOException {
+  public synchronized long bytesWritten() throws IOException {
```

Comment:
`FsDataWrtiter` does not have `bytesWritten`. `SimpleDataWriter.bytesWritten()` doesn't need synchronization because it doesn't check the output file.